runner/src/simulate.rs

```diff
@@ -205,6 +205,111 @@ impl MetricsCollection for PromMetricCollector {
     }
 }
 
+#[derive(Debug, Clone)]
+struct PeerRequestInfo {
```

Comment:
Just a name comment, I feel like we should have Metric in the name or something to distinguish this from Peer Requesting Info 😵 

PeerRequestMetricInfo?